libarchive: Backport patch from upstream (rev 2521).

Disable dot stripping code since it's still broken
and noone has been able to figure it out (yet).

Upstream-Status: Backport

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

diff --git a/libarchive/archive_read_support_format_iso9660.c b/libarchive/archive_read_support_format_iso9660.c
index 2d3a855..8661532 100644
--- a/libarchive/archive_read_support_format_iso9660.c
+++ b/libarchive/archive_read_support_format_iso9660.c
@@ -1771,9 +1771,11 @@ parse_file_info(struct archive_read *a, struct file_info *parent,
 			*wp = L'\0';
 		}
 
+#if 0 /* XXX: this somehow manages to strip of single-character file extensions, like '.c'. */
 		/* Chop off trailing '.' from filenames. */
 		if (*(wp-1) == L'.')
 			*(--wp) = L'\0';
+#endif
 
 		/* store the result in the file name field. */
 		archive_strappend_w_utf8(&file->name, wbuff);
-- 
1.7.1

